src/wovnio/wovnphp/core/WovnRequest.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+
+namespace Wovnio\Wovnphp\Core;
+
+require_once(__DIR__ . '/WovnLangDirectory.php');
+require_once(__DIR__ . '/WovnLang.php');
+require_once(__DIR__ . '/WovnOption.php');
+require_once(__DIR__ . '/WovnURL.php');
+
+
+class WovnRequest
+{
+    private $scheme; // e.g. HTTP
+    private $host; // e.g. 20.18.166.31, news.network.com, news.com
+    private $path; // e.g. /books/bestseller/1.html, /
+    private $url; // The fully qualified request URL, as-is, e.g. https://www.new.com/breaking/top.html?start=1&wovn=en
+    private $wovnUrl; // The WovnURL object
+    private $query; // The request's query, as an associative array
+    private $lang; // The request's target language code
+
+    private $langDirectory;
+    private $options;
+
+    public function __construct($serverSuperGlobal, $optionConfig, $useStrictMode = true)
+    {
+        $this->options = new WovnOption($optionConfig, $useStrictMode); // Do Not handle exceptions here, let them fail.
+        $this->scheme = $this->parseScheme($serverSuperGlobal);
+        $this->host = $serverSuperGlobal['HTTP_HOST'];
+        $this->path = $this->parseURI($serverSuperGlobal);
+        $this->query = isset($serverSuperGlobal['QUERY_STRING']) ? $serverSuperGlobal['QUERY_STRING'] : '';
+        $this->url = $this->scheme . '://' . $this->host . $this->path . $this->query;
+        $this->langDirectory = new WovnLangDirectory(
+            $this->options->get(WovnOption::OPT_SUPPORTED_LANGS),
+            $this->options->get(WovnOption::OPT_DEFAULT_LANG),
+            $this->options->get(WovnOption::OPT_CUSTOM_LANG_ALIASES)
+        );
+        $this->wovnUrl = new WovnURL($this->url, $this->langDirectory, $this->options);
+        $this->lang = $this->wovnUrl->lang();
+    }
+
+    /**
+     * Swaps the HTML either locally or via API, depending on the situation
+     */
+    public function swap()
```

Comment:
i'm not sure what functionality is planned to be in here, but it feels beyond the scope of this class. It would be better to keep this class just related to telling you about the state of the current request, what language has been calculated for it etc.

Taking the response body/headers and translating it should be done by something else.

Reply:
The WovnRequest class is supposed to be the "entry point" for integration, as this will allow the maximum amount of abstraction (you only need to provide a $_SERVER superglobal and the wovn configuration). And the swap() function is exposed to the developers as the only public function (right now lang() and wovnUrl() are exposed to help with partial integration). The idea is to just instantiate WovnRequest object and call the swap() function, then you get a translated or original HTML, with the server headers already modified for you, ready to be sent out.

Reply:
OK, well I would tend more towards a name like `translate`.  "swapping" is just one part of what these libraries do